### PR TITLE
[812] Fixing import visibility in Deresolver

### PR DESCRIPTION
**Pull request by ebausson-obeo**
Bug: https://github.com/eclipse-syson/syson/issues/812

# PLEASE READ ALL ITEMS AND CHECK ONLY RELEVANT CHECKBOXES BELOW

## Project management

- [x] Has the pull request been added to the relevant milestone?
- [x] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [x] Have the relevant issues been added to the pull request?
- [x] Have the relevant labels been added to the issues? (`area:`, `type:`)
- [x] Have the relevant issues been added to the same project milestone as the pull request?

## Changelog and release notes

- [x] Has the `CHANGELOG.adoc` + `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc` + `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc`?
- [ ] In case of a change with a visual impact, are there any screenshots in the `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc`?
- [ ] In case of a key change, has the change been added to `Key highlights` section in `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc`?
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc` + `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc`?

## Documentation

- [ ] Have you included an update of the [documentation](https://doc.mbse-syson.org) in your pull request? Please ask yourself if an update (installation manual, user manual, developer manual...) is needed and add one accordingly.

## Tests

- [ ] Is the code properly tested? Any pull request (fix, enhancement or new feature) should come with a test (or several). It could be unit tests, integration tests or cypress tests depending on the context. Only doc and releng pull request do not need for tests.

### CODE REVIEW

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/utils/NameDeresolver.java [hunk 1]
```diff
@@ -140,7 +141,8 @@ private String deresolve(Element element, Namespace sourceNamespace, Namespace d
             if (cacheValue != null) {
                 return cacheValue;
             }
-            EList<Membership> visibleMemberships = this.getVisibleMemberships(deresolvingNamespace, deresolvingNamespace == sourceNamespace);
+            boolean includePrivate = deresolvingNamespace == sourceNamespace || EcoreUtil.isAncestor(deresolvingNamespace, sourceNamespace);
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/utils/NameDeresolver.java:144]: Do not add boolean expressions with the following format: `x == y || z`.
It's not readable.
Please break operands down, at least in two.

File: backend/application/syson-application/src/test/java/org/eclipse/syson/application/export/ImportExportTests.java [hunk 2]
```diff
@@ -132,4 +132,58 @@ public void checkUseCaseTest() throws IOException {
...
 
+    /**
+     * Test import/export on test file importTest.sysml.
+     *
+     * @see https://github.com/Systems-Modeling/SysML-v2-Release/blob/master/sysml/src/examples/Simple%20Tests/ImportTest.sysml
+     */
+    @Test
+    public void checkImportTest() throws IOException {
+        /*
+         * The file has been modified because a problem has been detected during the export phase that force us to use
```
Comment (AxelRICHARD) [inline backend/application/syson-application/src/test/java/org/eclipse/syson/application/export/ImportExportTests.java:143]: Could you be more precise on the problem you mention here?
You said it should be investigated, but if someone (you or any other person) try to investigate in a few weeks/months, he will not able to start the investigation without more context.

Comment (ebausson-obeo) [reply backend/application/syson-application/src/test/java/org/eclipse/syson/application/export/ImportExportTests.java]: done.

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/utils/NameDeresolver.java (hunk 1 above)
Comment (ebausson-obeo) [reply backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/utils/NameDeresolver.java]: done.

**Review by AxelRICHARD — APPROVED** [review]

**Pull request merged**